pkg/daemon/loop.go

```diff
@@ -70,7 +76,21 @@ func (d *Daemon) Loop(stop chan struct{}, wg *sync.WaitGroup, logger log.Logger)
 	d.AskForSync()
 	d.AskForAutomatedWorkloadImageUpdates()
 
+	// How long until we consider a loop to be "stuck", in seconds, for
+	// failing liveness checks, which is given to be the smallest interval
+	// + corresponding timeout, and no less than 1 minute.
+	loopTimeout := (d.SyncInterval.Milliseconds() + d.SyncTimeout.Milliseconds()) / 1000
+	// FIXME: What timeout should we use for AutomationInterval?
+	if x := (d.AutomationInterval.Milliseconds() * 2) / 1000; x < loopTimeout {
```

Comment:
I wasn't sure what timeout to use for this, so I just set it to double the interval.